.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,237 @@
+# Pre-requisites:
+# - AWS Account with correct permissions
+# - `gh-pages` branch
+# - AWS Account needs to add the LOG_GROUP_NAME defined below
+
+name: Soak Testing
+on:
+  workflow_dispatch:
+    inputs:
+      target_commit_sha:
+        description: 'The commit SHA on this repo to use for the Soak Tests.'
+        required: true
+      test_duration_minutes:
+        description: 'The duration of the Soak Tests in minutes.'
+        required: true
+        default: 300
+  schedule:
+    - cron: '0 15 * * *'
+env:
+  # NOTE: The configuration of `APP_PROCESS_EXECUTABLE_NAME` is repo dependent
+  APP_PROCESS_EXECUTABLE_NAME: ruby
+  AWS_DEFAULT_REGION: us-east-1
+  DEFAULT_TEST_DURATION_MINUTES: 300
+  HOSTMETRICS_INTERVAL_SECS: 600
+  CPU_LOAD_THRESHOLD: 30
+  TOTAL_MEMORY_THRESHOLD: 1073741824 # 1 GiB
+  MAX_BENCHMARKS_TO_KEEP: 100
+  LISTEN_ADDRESS_PORT: 8080
+  # TODO: We might be able to adapt the "Soak Tests" to be "Overhead Tests".
+  # This means monitoring the Sample App's performance using high levels of TPS
+  # for the Load Generator over a shorter period of testing time. For example:
+  # https://github.com/aws-observability/aws-otel-collector/blob/main/docs/performance_model.md
+  # THROUGHPUT_PER_SECOND: TBD?
+
+jobs:
+  test_apps_and_publish_results:
+    name: Soak Performance Test - (${{ matrix.app-platform }}, ${{ matrix.instrumentation-type }})
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
+      issues: write
+    strategy:
+      fail-fast: false
+      matrix:
+        app-platform: [ ruby-on-rails ]
+        instrumentation-type: [ manual ]
+    env:
+      # NOTE: The configuration of `APP_PATH` is repo dependent
+      APP_PATH: sample-apps/${{ matrix.instrumentation-type}}-instrumentation/${{ matrix.app-platform }}
+      LOGS_NAMESPACE: ${{ github.repository }}/soak-tests-${{ matrix.app-platform }}-${{ matrix.instrumentation-type }}
+    steps:
+    # MARK: - GitHub Workflow Event Type Specific Values
+
+      - name: Use INPUT as commit SHA
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        run: |
+          echo "TARGET_SHA=${{ github.event.inputs.target_commit_sha }}" | tee --append $GITHUB_ENV;
+      - name: Use LATEST as commit SHA
+        if: ${{ github.event_name != 'workflow_dispatch' }}
```

Comment:
Could probably just specify the `schedule` event.

Reply:
I think this has the benefit of reading as "if A else" because we say "if not A". I just don't want us to add another trigger (like `synchronize`) and not have this set because we forgot to do `== schedule || == synchronize`.

I will say that I am a little hesitant to change code in this PR for minor changes 😓 Because this code is in 5 different repos and changes here will be confusing unless we fix it everywhere. We'll definitely improve it later when we centralize them so we only have to change repo at a time 🙂